python/lsst/pipe/tasks/measurementDriver.py

```diff
@@ -0,0 +1,1171 @@
+            # N.B. Scarlet is designed to leverage multi-band information to
+            # differentiate overlapping sources based on their spectral and
+            # spatial profiles. However, it can also run on a single band and
+            # often give better results than 'meas_deblender'.
+            self.log.info(f"Running '{self._Deblender}' in single-band mode; make sure it was intended!")
+            if refBand is None:
+                refBand = mExposure.filters[0]
+                self.log.info(
+                    "No reference band provided for single-band data; "
+                    f"using the only available band ('{refBand}') as the reference band"
+                )
+        else:
+            if catalog is None:
+                if self.config.measureOnlyInRefBand:
+                    measInfo = "and everything downstream of deblending"
+                else:
+                    measInfo = (
+                        "while subtasks downstream of deblending will be run in each of "
+                        f"the {mExposure.filters} bands"
+                    )
+                self.log.info(f"Using '{refBand}' as the reference band for detection {measInfo}")
+
+        # Final sanity checks after all the adjustments above.
+        if refBand is None:
+            raise ValueError("Reference band must be provided for multi-band data")
+
+        if refBand not in mExposure.filters:
+            raise ValueError(f"Requested band '{refBand}' is not present in the multi-band exposure")
+
+        if bands is not None and refBand not in bands:
+            raise ValueError(f"Reference band '{refBand}' is not in the list of 'bands' provided: {bands}")
+
+        return mExposure, refBand, bands
+
+    def _deblendSources(
+        self, mExposure: afwImage.MultibandExposure, catalog: afwTable.SourceCatalog, refBand: str
+    ):
+        """Run multi-band deblending given a multi-band exposure and a catalog.
+
+        Parameters
+        ----------
+        mExposure :
+            Multi-band exposure on which to run the deblending algorithm.
+        catalog :
+            Catalog containing sources to be deblended.
+        refBand :
+            Reference band used for detection or the band to use for
+            measurements if `measureOnlyInRefBand` is enabled.
+
+        Returns
+        -------
+        catalogs :
+            Dictionary of catalogs containing the deblended sources. If
+            `measureOnlyInRefBand` is enabled, this will only contain the
+            reference-band catalog; otherwise, it will contain a catalog for
+            each band.
+        modelData :
+            Multiband scarlet models produced during deblending.
+        """
+        self.log.info(f"Deblending using '{self._Deblender}' on {len(catalog)} detection footprints")
+
+        # Run the deblender on the multi-band exposure.
+        catalog, modelData = self.deblend.run(mExposure, catalog)
```

Comment:
Please cache modelData (in other words `self.modelData = modelData`. This will be useful for debugging.

Reply:
Done here:
```python
if self.config.doDeblend:
    catalogs, self.modelData = self._deblendSources(mExposure, catalog, refBand=refBand)
```